### PR TITLE
Qs editor interfaces questionbank and test cases

### PR DESCRIPTION
**Pull request by JagadishPujari**

### CODE REVIEW

File: editor/questionbankcontroller.js [hunk 1]
```diff
@@ -182,6 +181,51 @@ angular.module('createquestionapp', [])
       $scope.searchQuestions();
       $scope.selectedIndex = undefined;
 
+      //Load all plugins
+      $scope.loadPlugins = function(plugins, manifestMedia) {
```
Comment (manojchandrashekar) [inline editor/questionbankcontroller.js]: Move function definition out of `$scope.init()`.

**Review by manojchandrashekar — CHANGES_REQUESTED** [review]

File: test/mocks/ediotor/eventListener-mock.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+windowmock.open = function(url, target, settings){
+
+    return {
+        addEventListener: function(event, callback){
+            if (event == 'loadstart'){
+                callback({
+                    url: 'something.com or whatever youre expecting',
```
Comment (manojchandrashekar) [inline test/mocks/ediotor/eventListener-mock.js]: What is this string?

**Review by manojchandrashekar — APPROVED** [review]

**Comment by manojchandrashekar** [comment]
@vinukumar-vs FYA

File: editor/questionbankcontroller.js [hunk 3]
```diff
@@ -62,8 +61,7 @@ angular.module('createquestionapp', [])
       }
     };
     $scope.csspath = ecEditor.resolvePluginResource(pluginInstance.manifest.id, pluginInstance.manifest.ver, 'editor/style.css');
-    $scope.contentNotFound = ecEditor.resolvePluginResource(pluginInstance.manifest.id, pluginInstance.manifest.ver, 'assets/contentnotfound.jpg');
-
+    $scope.contentNotFound = ecEditor.resolvePluginResource(pluginInstance.manifest.id, pluginInstance.manifest.ver, 'assets/contentnotfound.jpg');
```
Comment (vinukumar-vs) [inline editor/questionbankcontroller.js]: Naming convention is not correct. It should be **noQuestionFound**

File: editor/questionbankcontroller.js [hunk 4]
```diff
@@ -91,6 +89,7 @@ angular.module('createquestionapp', [])
     });
 
     $scope.searchQuestions = function (filterData, callback) {
+      $scope.itemsLoading = true;
```
Comment (vinukumar-vs) [inline editor/questionbankcontroller.js]: It should be "true" only when we are searching questions(only 1 place). It should hide after getting response(only 1 place). 

Remove from other unwanted places.

File: editor/questionbankcontroller.js [hunk 5]
```diff
@@ -172,6 +171,19 @@ angular.module('createquestionapp', [])
...
+    //Load all plugins
+    $scope.loadPlugins = function(plugins, manifestMedia) {
+      var pluginObj = [];
+      if (!Array.isArray(plugins)) {
+        pluginObj.push(plugins);
+        plugins = pluginObj;
+      }
+      org.ekstep.pluginframework.pluginManager.loadAllPlugins(plugins, manifestMedia, function() {
+        if (typeof PluginManager != 'undefined') {
+          PluginManager.pluginMap = org.ekstep.pluginframework.pluginManager.plugins;
```
Comment (vinukumar-vs) [inline editor/questionbankcontroller.js]: Why this is implemented? I think this has already taken care by plugin-framework.

File: editor/questionbankcontroller.js [hunk 6]
```diff
@@ -181,6 +193,37 @@ angular.module('createquestionapp', [])
...
+            "targets.ver": {'<=': Number(qsVesrion)},
+            "status": "Live"
+          },
+          "limit": 50,
+          "fields": ['contentType','semanticVersion','appIcon']
+        }
+      };
+      ecEditor.getService('search').search(data, function(err, resp) {
+        var pluginsData = resp.data.result.content;
+        localStorage.setItem("qs-plugins", JSON.stringify(pluginsData));
```
Comment (vinukumar-vs) [inline editor/questionbankcontroller.js]: Why are you using localStorage? Can't we go with singleton class object store in framework.

File: editor/questionbankcontroller.js [hunk 7]
```diff
@@ -181,6 +193,37 @@ angular.module('createquestionapp', [])
...
+          if (value) {
+            var obj = {
+              "id": value.identifier,
+              "ver": value.semanticVersion,
+              "type": 'plugin'
+            }
+            plugins.push(obj);
+          }
+        });
+        $scope.loadPlugins(plugins, []);
```
Comment (vinukumar-vs) [inline editor/questionbankcontroller.js]: Why are loading the plugin using scope level function? You can loadPlugin by calling ContentEditor API itself.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: editor/questionbankcontroller.js (hunk 5 above)
Comment (JagadishPujari) [reply editor/questionbankcontroller.js]: Question unit (MCQ/MTF/FTB)plugins not loading by plugin framework, So we are getting QS related units plugins by services and loading implicitly in the Question bank.

File: editor/questionbankcontroller.js (hunk 6 above)
Comment (JagadishPujari) [reply editor/questionbankcontroller.js]: Just storing which all implicitly loaded plugins in localStorage and after getting data we are clearing in question plugin.

Comment (vinukumar-vs) [reply editor/questionbankcontroller.js]: This should be plugin.js functionality, not the controller functionality. QuestionSet plugin has to load its dependencies.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**